app/services/service_provider_updater.rb

```diff
@@ -0,0 +1,47 @@
+class ServiceProviderUpdater
+  def run
+    dashboard_service_providers.each do |sp|
+      issuer = sp['issuer'] || sp[:issuer]
+      SERVICE_PROVIDERS[issuer] = sp
+
+      VALID_SERVICE_PROVIDERS << issuer unless VALID_SERVICE_PROVIDERS.include?(issuer)
+
+      # must also alter secure headers
+      # see config/initializers/secure_headers.rb
+      alter_secure_headers(sp)
+    end
+  end
+
+  private
+
+  def url
+    Figaro.env.dashboard_url
+  end
+
+  def dashboard_service_providers
+    body = dashboard_response.body
+    return JSON.parse(body) if dashboard_response.code == 200
+    log_error "Failed to parse response from #{url}: #{body}"
+    []
+  rescue
+    log_error "Failed to contact #{url}"
+    []
+  end
+
+  def dashboard_response
+    @_dashboard_response ||= HTTParty.get(url)
+  end
+
+  def alter_secure_headers(sp)
```

Comment:
is this so we can allow non-https service providers? 

That is an issue that I ran into this morning when trying to test a service provide running localhost against our idp on dev. It worked on Firefox, but not chrome (annoyingly, there was no helpful error message, but I did see this in the js console:

![screen shot 2017-01-18 at 10 37 02 am](https://cloud.githubusercontent.com/assets/601515/22089356/6f53d1d8-dd9f-11e6-9f1a-2c207fbc0f4c.png)

If this is not a fix, what does it do?

Either way, would there be a way to test this? I think a test would clarify what's happening here